### PR TITLE
[Merged by Bors] - docs: improve `Finite` docstring

### PR DESCRIPTION
**Pull request by kmill**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/Finite/Defs.lean [hunk 1]
```diff
@@ -50,11 +50,32 @@ open Function
...
+and use this `Fintype.ofFinite` interface.
+The `Fintype` instances should be relied upon to be computable for evaluation purposes.
+
+Theorems should use `Finite` instead of `Fintype`, unless definitions in the theorem statement
+require `Fintype`.
+Definitions should prefer `Finite` as well, unless it is important that the definitions
+are meant to be computable in the reduction or `#eval` sense.
+
+Implementation note: while this could be defined as `Nonempty (Fintype α)`,
+it is defined in the current way to allow there to be `Finite p` instances for `p : Prop`.
```
Comment (eric-wieser) [inline Mathlib/Data/Finite/Defs.lean:78]: The last half of this last line is not just an implementation note; the fact that `Finite` works on propositions is a user-visible feature, and very useful for working with indexed finite collections, such as when using `Finsum`.

Comment (kmill) [reply Mathlib/Data/Finite/Defs.lean:78]: Ok, how about "design note"?

Comment (eric-wieser) [reply Mathlib/Data/Finite/Defs.lean:78]: I think it's worth having near the top: Finite is a) itself prop not data, and b) works on proof types as well a data types. Both are important API features, and often when you generalize a theorem for a) you should also try to do it for b)

Comment (kmill) [reply Mathlib/Data/Finite/Defs.lean]: Ok, updated

**Comment by kmill** [comment]
@eric-wieser Are we happy to merge this yet?

**Comment by j-loreaux** [comment]
LGTM

bors d=eric-wieser

**Review by eric-wieser — COMMENTED** [review]
bors merge

**Pull request closed without merging**